activiti-spring-boot-starter/src/test/resources/processes/IntermediateTimerEventTest.testCatchingTimerEvent.bpmn20.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<definitions id="definitions" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:omgdc="http://www.omg.org/spec/DD/20100524/DC" xmlns:omgdi="http://www.omg.org/spec/DD/20100524/DI"
+             xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL"
+             xmlns:activiti="http://activiti.org/bpmn"
+             targetNamespace="Examples">
+
+    <process id="intermediateTimerEventExample" name="Timer intermediate event example">
+
+        <startEvent id="theStart"/>
+        <sequenceFlow id="flow1" sourceRef="theStart" targetRef="timer"/>
+
+        <intermediateCatchEvent id="timer">
+            <timerEventDefinition>
+                <timeDuration>PT5M</timeDuration>
+            </timerEventDefinition>
+        </intermediateCatchEvent>
+
+        <sequenceFlow id="flow2" sourceRef="timer" targetRef="theEnd"/>
```

Comment:
Maybe it's better to add a user task between timer and end event, so you can test that after the timer the execution reaches the user task.

Reply:
Test updated: https://github.com/Activiti/Activiti/pull/2814/commits/c8e0c288fc269f4417d85e20a34b5d969801775c
@erdemedeiros 
You may see here 
https://github.com/Activiti/Activiti/blob/c8666252c669d89f2c6d7063e4a89aadf6b73175/activiti-engine/src/test/java/org/activiti/engine/test/api/event/JobEventsTest.java#L299
that additionally to implemented events we will get also following events (related to timers):
`
   // timer entity created first
    assertEquals(ActivitiEventType.ENTITY_CREATED, listener.getEventsReceived().get(0).getType());
    // timer entity initialized
    assertEquals(ActivitiEventType.ENTITY_INITIALIZED, listener.getEventsReceived().get(1).getType());
    // timer entity deleted
    assertEquals(ActivitiEventType.ENTITY_DELETED, listener.getEventsReceived().get(2).getType());
  `